### PR TITLE
Caching Least Recently Used Contract Code in Stage Execution

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

**Comment by Giulio2002** [comment]
Basically, there is a series of buckets who is considered "whitelisted", which means it wont be expected to change much... when we read a non-batched value from these buckets we cache it an `*lru.Cache`

File: eth/stagedsync/stage_execute.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"time"
 
 	"github.com/c2h5oh/datasize"
+	lru "github.com/hashicorp/golang-lru"
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_execute.go:12]: you have global lock in mutation - then you don't need one more lock inside lru (probably), then you can use `simplelru` sub-package.

File: eth/stagedsync/stage_execute.go [hunk 2]
```diff
@@ -36,7 +37,8 @@ import (
 )
 
 const (
-	logInterval = 30 * time.Second
+	logInterval    = 30 * time.Second
+	lruDefaultSize = 1_000_000
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_execute.go:41]: add some comment - how much is it in Mb/Gb of RAM? Some empiric estimation.

File: eth/stagedsync/stage_execute.go (hunk 1 above)
Comment (AskAlexSharov) [reply eth/stagedsync/stage_execute.go:12]: there is also github.com/VictoriaMetrics/fastcache package which can be more GC-friendly, but I'm not sure.

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**